File: codeforces/contests/21/c.py
import sys
import threading
from collections import defaultdict

input = lambda: sys.stdin.readline().strip()

"""
Rumor

https://codeforces.com/gym/444629/problem/C

Time - O(nlog(n))
Space - O(n)
"""

def dfs(d: defaultdict, visited: set, i: int):
    if i in visited:
        return

    visited.add(i)
    if i in d:
        for c in d[i]:
            dfs(d, visited, c)


def main():
    n, m = map(int, input().split())
    c = list(map(int, input().split()))

    d = defaultdict(list)

    for _ in range(m):
        x, y = map(int, input().split())

        d[x].append(y)
        d[y].append(x)

    min_bribe = sorted([(c[i], i) for i in range(n)])
    visited = set()

    res = 0
    if m == 0:
        res = sum(c)
    else:
        for b, i in min_bribe:
            if i + 1 not in visited:
                dfs(d, visited, i + 1)
                res += b

    print(res)


if __name__ == "__main__":
    sys.setrecursionlimit(1 << 30)
    threading.stack_size(1 << 27)

    main_thread = threading.Thread(target=main)
    main_thread.start()
    main_thread.join()

File: codeforces/contests/21/c.cpp
#include <bits/stdc++.h>

using namespace std;

/**
 * XOR Mixup
 * https://codeforces.com/gym/443056/problem/C
 *
 * Time - O(n^2)
 * Space - O(1)
 */

int main()
{
    int t, n, res;
    vector<int> a;

    ios::sync_with_stdio(false);
    cin.tie(0);

    cin >> t;

    for (auto _ = 0; _ < t; ++_) {
        cin >> n;

        a.resize(n);
        for (auto i = 0; i < n; ++i) cin >> a[i];

        /**
         * because
         * a[0] ^ a[1] ^ ...a[n - 1] = 0
         * implies
         * a[0] ^ a[1] ^ ...a[n-1] ^ a[n - 1] = a[0] ^ a[1] ^ ...a[n-2] = a[n - 1]
         */
        cout << a[0] << "\n";
    }

    return 0;
}
